lib/cloud_pub_sub.ex

```diff
@@ -34,6 +34,8 @@ defmodule CloudPubSub do
   defdelegate connected?(), to: CloudPubSub.Adapter
 
   defdelegate publish(topic, payload, opts), to: CloudPubSub.Adapter
+  defdelegate publish_cast(topic, payload, opts), to: CloudPubSub.Adapter
```

Comment:
>This prevents deadlocks

Out of curiosity, have you tried `qos: 0`? I suspect the default QOS 1 isn't sending an ACK until the handler returns, which may be causing the deadlock.

Reply:
Yes I have. The problem is that the function is a call, so someone cannot publish or prepare to publish from the PubSubHandler callbacks since the Adapter is still waiting for the handler to return. 
The cast allows me to drop a publish into the Adapters mailbox and then return control back to it.